sdk/cognitiveservices/azopenai/custom_client.go

```diff
@@ -0,0 +1,155 @@
+	if options == nil {
+		options = &ClientOptions{}
+	}
+	if deploymentID == "" {
+		return nil, errors.New("parameter client.deploymentID cannot be empty")
+	}
+	authPolicy := NewAPIKeyPolicy(credential, "api-key")
+	azcoreClient, err := azcore.NewClient(clientName, version, runtime.PipelineOptions{PerRetry: []policy.Policy{authPolicy}}, &options.ClientOptions)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{endpoint: endpoint + "/openai", deploymentID: deploymentID, internal: azcoreClient}, nil
+}
+
+// NewClientForOpenAI creates a new instance of Client with the specified values.
+//   - endpoint - OpenAI service endpoint
+//   - credential - used to authorize requests with an API Key credential
+//   - options - client options, pass nil to accept the default values.
+func NewClientForOpenAI(endpoint string, credential KeyCredential, options *ClientOptions) (*Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+	openAIPolicy := newOpenAIPolicy(credential)
+	azcoreClient, err := azcore.NewClient(clientName, version, runtime.PipelineOptions{PerRetry: []policy.Policy{openAIPolicy}}, &options.ClientOptions)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{endpoint: endpoint, internal: azcoreClient}, nil
+}
+
+// openAIPolicy is an internal pipeline policy to remove the api-version query parameter
+type openAIPolicy struct {
+	cred KeyCredential
+}
+
+// newOpenAIPolicy creates a new instance of openAIPolicy.
+// cred: a KeyCredential implementation.
+func newOpenAIPolicy(cred KeyCredential) *openAIPolicy {
+	return &openAIPolicy{cred: cred}
+}
+
+// Do returns a function which adapts a request to target OpenAI.
+// Specifically, it removes the api-version query parameter.
+func (b *openAIPolicy) Do(req *policy.Request) (*http.Response, error) {
+	q := req.Raw().URL.Query()
+	q.Del("api-version")
+	req.Raw().Header.Set("authorization", "Bearer "+b.cred.APIKey)
+	return req.Next()
+}
+
+// Methods that return streaming response
+
+type streamCompletionsOptions struct {
+	CompletionsOptions
+	Stream bool `json:"stream"`
+}
+
+func (o streamCompletionsOptions) MarshalJSON() ([]byte, error) {
+	bytes, err := o.CompletionsOptions.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	objectMap := make(map[string]any)
+	err = json.Unmarshal(bytes, &objectMap)
+	if err != nil {
+		return nil, err
+	}
+	objectMap["stream"] = o.Stream
+	return json.Marshal(objectMap)
+}
+
+// GetCompletionsStream - Return the completions for a given prompt as a sequence of events.
+// If the operation fails it returns an *azcore.ResponseError type.
+//   - options - ClientGetCompletionsOptions contains the optional parameters for the Client.GetCompletions method.
+func (client *Client) GetCompletionsStream(ctx context.Context, body CompletionsOptions, options *ClientGetCompletionsStreamOptions) (CompletionEventsResponse, error) {
```

Comment:
I believe this is what you're looking for:

https://github.com/mikekistler/azure-sdk-for-go/blob/15ed993832fae85a7158006a3b0c4a3f4e17d035/sdk/cognitiveservices/azopenai/custom_models.go#L17

I can also just file an issue for it since we're at a good merge point now.

Reply:
Thanks, please change to:
// GetCompletionsStreamResponse is the response from [GetCompletionsStream].
type GetCompletionsStreamResponse struct {
	// CompletionsStream returns the stream of completions. Token limits and other settings may limit the number of completions returned by the service.
	CompletionsStream *EventReader[Completions]
}

Also, please make sure we have an example of how to use this operation in the docs.

Reply:
API fixed, writing the example now.

Reply:
Added.